web/impact/impact/v1/helpers/criterion_helper.py

```diff
@@ -32,15 +32,30 @@ class CriterionHelper(ModelHelper):
     INPUT_KEYS = ALL_KEYS
 
     specific_helpers = {}
+    helper_instances = {}
+    criterion_total_capacities = {}
+
+    def __init__(self, subject):
+        super().__init__(subject)
+        self.app_count_cache = 0
 
     @classmethod
     def register_helper(cls, klass, type, name):
         cls.specific_helpers[(type, name)] = klass
 
     @classmethod
     def find_helper(cls, criterion):
-        return cls.specific_helpers.get((criterion.type, criterion.name),
-                                        cls)(criterion)
+        criteria_tuple = (criterion.type, criterion.name)
+        if cls.helper_instances.get(criteria_tuple) is None:
+            cls.helper_instances[criteria_tuple] = cls.specific_helpers.get(
+                criteria_tuple,
+                cls
+            )(criterion)
+        return cls.helper_instances[criteria_tuple]
+
+    @classmethod
+    def clear_cache(cls):
+        cls.helper_instances = {}
```

Comment:
does self. app_count_cache need to be reset here?

Reply:
this cache is at the instance level, so it get's destroyed at the end of the request by Djangos garbage collector